### PR TITLE
doc: zbus: samples: Use new Sphinx extension to document samples

### PR DESCRIPTION
**Pull request by kartben**
Use the new code-sample directive and roles to document the zbus samples so that they show up as "Related samples" when browsing zbus API.

### CODE REVIEW

File: samples/subsys/zbus/hello_world/README.rst [hunk 1]
```diff
@@ -1,7 +1,8 @@
-.. _zbus-hello-world-sample:
+.. zephyr:code-sample:: zbus-hello-world
+   :name: zbus Hello World
+   :relevant-api: zbus_apis
 
-Hello world sample
-##################
+   Make two threads talk to each other using zbus.
```
Comment (rodrigopex) [inline samples/subsys/zbus/hello_world/README.rst:5]:
```suggestion
   Make three threads talk to each other using zbus.
```

Comment (rodrigopex) [reply samples/subsys/zbus/hello_world/README.rst:5]: The main thread is involved in the conversation.

**Review by rodrigopex — CHANGES_REQUESTED** [review]
Only small things to adjust. Great summaries! Thanks for the PR.

**Comment by kartben** [comment]
> Only small things to adjust. Great summaries! Thanks for the PR.

thanks for the quick review! should be good now :) live result should eventually be visible here: https://builds.zephyrproject.io/zephyr/pr/62360/docs/services/zbus/index.html#api-reference

**Review by fabiobaltieri — APPROVED** [review]

**Comment by kartben** [comment]
@rodrigopex please revisit when you have a chance :)

**Review by rodrigopex — APPROVED** [review]

**Pull request merged**